zmq.hpp

```diff
@@ -615,8 +620,340 @@ inline void swap(context_t &a, context_t &b) ZMQ_NOTHROW {
     a.swap(b);
 }
 
+#ifdef ZMQ_CPP11
+struct send_result
+{
+    size_t size;    // message size in bytes
+    bool success;
+};
+
+struct recv_result
+{
+    size_t size;    // message size in bytes
+    bool success;
+};
+
+struct recv_buffer_result
+{
+    size_t size;    // number of bytes written to buffer
+    size_t untruncated_size;  // untruncated message size in bytes
+    bool success;
+
+    ZMQ_NODISCARD bool truncated() const noexcept
+    {
+        return size != untruncated_size;
+    }
+};
+
+enum class send_flags : int
```

Comment:
I am not 100% sure if an enum should be used for flags. 

While IIRC the C & C++ standards specifically allow this, I find it quite surprising that there are possible values that are not listed in the enum declaration. On the other hand, the enum name indicates that these are flags, so the probability of confusion would be acceptable.

However, unfortunately, there is no perfect solution for this in C++. An alternative would be to use a bitfield for this purpose. What would you think about that? 

In any case, this should be done uniformly with cppzmq, and already your other open PR also introduces a flags enum, so we should take a moment to be sure this is the way to go. It would also be good to add some documentation on such general conventions, so a future discussion can reference that. Maybe an additional markdown file CONVENTIONS.md could be added for that.

Reply:
The standard library has a BitmaskType requirement https://en.cppreference.com/w/cpp/named_req/BitmaskType . The latest bitmask additions to the library (`std::launch` for `std::async` and those in `std::filesystem`) are all specified as `enum class`, so I believe that convention is quite common for those using newer standards.
I could also comment in code that those flags satisfy (partially, not all bit operations are implemented) the named requirement BitmaskType, would that be satisfactory?

Reply:
Ok, wasn't aware of that the standard library does it like that as well. Then leave it as it is :) Maybe add the comment as suggested, but it is not essential.